### ISSUE TITLE
Accessibility Issues in Navigation Sidebar

### ISSUE THREAD

**Issue by kayla-beckham** [scalar/scalar#3439]
### What happens?

![Screen Shot 2024-10-10 at 9 15 40 AM](https://github.com/user-attachments/assets/6ac5b08b-6e07-45d5-872b-ff0ee1a3140e)

1. The table of contents element is not announced by the screen reader. It should have an aria-label
2. In the table of contents, the screen reader does not announce a the state of the collapsable button. Need to add aria-expanded=true/false so that the state is announced by screen reader
    1. The toggle button for each section should have a meaningful aria-label that associates it with the section button. The screen reader should be able to announce ‘Expand <name of section>’, for example

### What did you expect to happen?

The context and state of elements in the navigation sidebar should be announced by a screen reader. The Sidebar itself should be announced by the screen reader.
